tutorials/clinical/10_ieeg_localize.py

```diff
@@ -0,0 +1,505 @@
+ch_names = elec_df['name'].tolist()
+ch_coords = elec_df[['R', 'A', 'S']].to_numpy(dtype=float)
+
+# load the subject's brain
+subject_brain = nib.load(op.join(misc_path, 'seeg', 'sample_seeg_brain.mgz'))
+
+# Make brain surface from T1
+verts, triangles = mne.marching_cubes(subject_brain.get_fdata(), level=100)
+# transform from voxels to surface RAS
+verts = mne.transforms.apply_trans(
+    subject_brain.header.get_vox2ras_tkr(), verts)
+
+renderer = mne.viz.backends.renderer.create_3d_figure(
+    size=(1200, 900), bgcolor='w', scene=False)
+mne.viz.set_3d_view(figure=renderer.figure, distance=700,
+                    azimuth=40, elevation=60, focalpoint=(0., 0., -45.))
+renderer.mesh(*verts.T, triangles=triangles, color='gray',
+              opacity=0.05, representation='surface')
+for ch_coord in ch_coords:
+    renderer.sphere(center=tuple(ch_coord), color='red', scale=5)
+
+renderer.show()
+
+###############################################################################
+# Warping to a Common Atlas
+# =========================
+#
+# Electrode contact locations are often compared across subjects in a template
+# space such as ``fsaverage`` or ``cvs_avg35_inMNI152``. To transform electrode
+# contact locations to that space, we need to determine a function that maps
+# from the subject's brain to the template brain. We will use the symmetric
+# diffeomorphic registration (SDR) implemented by ``Dipy`` to do this.
+#
+# Before we can make a function to account for individual differences in the
+# shape and size of brain areas, we need to fix the alignment of the brains.
+# The plot below shows that they are not yet aligned.
+
+# load the freesurfer average brain
+template_brain = nib.load(
+    op.join(subjects_dir, 'fsaverage', 'mri', 'brain.mgz'))
+
+plot_overlay(template_brain, subject_brain,
+             'Alignment with fsaverage before Affine Registration')
+
+###############################################################################
+# Now, we'll register the affine of the subject's brain to the template brain.
+# This aligns the two brains, preparing the subject's brain to be warped
+# to the template.
+
+# normalize intensities
+mri_to = template_brain.get_fdata().copy()
+mri_to /= mri_to.max()
+mri_from = subject_brain.get_fdata().copy()
+mri_from /= mri_from.max()
+
+# downsample for speed
+zooms = (5, 5, 5)
+mri_to, affine_to = reslice(
+    mri_to, affine=template_brain.affine,
+    zooms=template_brain.header.get_zooms()[:3], new_zooms=zooms)
+mri_from, affine_from = reslice(
+    mri_from, affine=subject_brain.affine,
+    zooms=subject_brain.header.get_zooms()[:3], new_zooms=zooms)
+
+reg_affine = affine_registration(
+    moving=mri_from,
+    static=mri_to,
+    moving_affine=affine_from,
+    static_affine=affine_to,
+    nbins=32,
+    metric='MI',
+    pipeline=[center_of_mass, translation, rigid, affine],
+    level_iters=[100, 100, 10],
+    sigmas=[3.0, 1.0, 0.0],
+    factors=[4, 2, 1])[1]
```

Comment:
This code is *very* similar to what we have with the code in `mne.morph._compute_morph_sdr`, and it's also referenced above. I think it's worth exposing some public function that calls that function internally to simplify this (the public version would probably only be a few lines at most). Maybe `mne.morph.align_volumes(img_from, img_to, zooms=None, n_iter_sdr=(...), ...)` and the call here would be something like:
```
affreg, _ = mne.morph.align_volumes(
    subject_brain, template_brain, zooms=(5, 5, 5),  # zooms != 1 here just for speed
    n_sdr_iter=())  # skip SDR step just for speed
reg_affine = affreg.affine
```
The second output here would be the SDR morph, which is None because `n_sdr_iter=()` here. And below we just say the complete call that was used actually didn't set `n_sdr_iter`, and is skipped here for speed because it takes too long. This is kind of where I was headed with the refactoring of `mne.morph._compute_morph_sdr`, so it shouldn't require too much code to get there I don't think. (The logging provided by that, including the R^2 values, is really nice, too, I think.)

But does it really still take too long when we use `zooms=(5, 5, 5)` and tweak the iterations? We use this in another example and it's fine:

https://github.com/mne-tools/mne-python/blob/653ada8b9b0d4937d0944afd5ad29712600892a5/examples/inverse/morph_volume_stc.py#L83-L87

Reply:
Yes, it still takes too long because those are downsampled by a factor of 5 and for accuracy, the SDR needs to be computed on the full-resolution brain although the affine registration can be on the downsampled brain.

Maybe this should be another PR to get this moving. Or if you want to refactor that would be good too but that's more things that might break and have to be fixed on the way and this is already getting quite extensive.

Reply:
Okay for now I'll just fix the tests I broke and then we can hopefully merge. Then I'll make a quick follow-up to finish the refactoring / new function idea. That's mostly separable anyway so will make reviewing easier anyway!